src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -177,97 +177,48 @@ struct HostFunctionMetrics
-            .Mark(mLedgerReadByte);
-        mMetrics.registry()
-            .NewMeter({"soroban", "host-fn-op", "read-data-byte"}, "byte")
-            .Mark(mReadDataByte);
-        mMetrics.registry()
-            .NewMeter({"soroban", "host-fn-op", "read-code-byte"}, "byte")
-            .Mark(mReadCodeByte);
-
-        mMetrics.registry()
-            .NewMeter({"soroban", "host-fn-op", "write-ledger-byte"}, "byte")
-            .Mark(mLedgerWriteByte);
-        mMetrics.registry()
-            .NewMeter({"soroban", "host-fn-op", "write-data-byte"}, "byte")
-            .Mark(mWriteDataByte);
-        mMetrics.registry()
-            .NewMeter({"soroban", "host-fn-op", "write-code-byte"}, "byte")
-            .Mark(mWriteCodeByte);
-
-        mMetrics.registry()
-            .NewMeter({"soroban", "host-fn-op", "emit-event"}, "event")
-            .Mark(mEmitEvent);
-        mMetrics.registry()
-            .NewMeter({"soroban", "host-fn-op", "emit-event-byte"}, "byte")
-            .Mark(mEmitEventByte);
-
-        mMetrics.registry()
-            .NewMeter({"soroban", "host-fn-op", "cpu-insn"}, "insn")
-            .Mark(mCpuInsn);
-        mMetrics.registry()
-            .NewMeter({"soroban", "host-fn-op", "mem-byte"}, "byte")
-            .Mark(mMemByte);
-        mMetrics.registry()
-            .NewTimer({"soroban", "host-fn-op", "invoke-time-nsecs"})
-            .Update(std::chrono::nanoseconds(mInvokeTimeNsecs));
-        mMetrics.registry()
-            .NewMeter({"soroban", "host-fn-op", "cpu-insn-excl-vm"}, "insn")
-            .Mark(mCpuInsnExclVm);
-        mMetrics.registry()
-            .NewTimer({"soroban", "host-fn-op", "invoke-time-nsecs-excl-vm"})
-            .Update(std::chrono::nanoseconds(mInvokeTimeNsecsExclVm));
-        mMetrics.registry()
-            .NewHistogram(
-                {"soroban", "host-fn-op", "invoke-time-fsecs-cpu-insn-ratio"})
-            .Update(mInvokeTimeNsecs * 1000000 /
-                    std::max(mCpuInsn, uint64_t(1)));
-        mMetrics.registry()
-            .NewHistogram({"soroban", "host-fn-op",
-                           "invoke-time-fsecs-cpu-insn-ratio-excl-vm"})
-            .Update(mInvokeTimeNsecsExclVm * 1000000 /
-                    std::max(mCpuInsnExclVm, uint64_t(1)));
-
-        mMetrics.registry()
-            .NewMeter({"soroban", "host-fn-op", "max-rw-key-byte"}, "byte")
-            .Mark(mMaxReadWriteKeyByte);
-        mMetrics.registry()
-            .NewMeter({"soroban", "host-fn-op", "max-rw-data-byte"}, "byte")
-            .Mark(mMaxReadWriteDataByte);
-        mMetrics.registry()
-            .NewMeter({"soroban", "host-fn-op", "max-rw-code-byte"}, "byte")
-            .Mark(mMaxReadWriteCodeByte);
-        mMetrics.registry()
-            .NewMeter({"soroban", "host-fn-op", "max-emit-event-byte"}, "byte")
-            .Mark(mMaxEmitEventByte);
+        mMetrics.mHostFnOpReadEntry.Mark(mReadEntry);
```

Comment:
With the change to init and cache at the ledger manager level, won't we end up accumulating the metrics across transactions in a ledger?

Reply:
The metrics you are pointing at is a meter, it always goes up. So even if we leave them in the operation, and call "NewMeter({...})" each time, it's going to use the same meter and accumulate that metric. 

When we plot the result on grafana, for example read entry per op, we have to do the subtractions ourselves, i.e. 
`increase(host_fn_op_read_entry[_interval]) / increase(host_fn_op_count[_interval])`

This works similarly for other metrics too. Medida contains an internal registry that caches and reuses metrics. Here we are just initializing them at the beginning, instead of lazily at each operation.